hw9/hw9_t03.py

```diff
@@ -0,0 +1,44 @@
+"""
+Write a function that takes directory path, a file extension and an optional tokenizer.
+
+It will count lines in all files with that extension if there are no tokenizer.
+If a the tokenizer is not none, it will count tokens.
+For dir with two files from hw1.py:
+universal_file_counter(test_dir, "txt")
+6
+universal_file_counter(test_dir, "txt", str.split)
+6
+"""
+import os
+from pathlib import Path
+from typing import Callable, Optional, Union
+
+
+def count_lines_or_tokens(
+    file_path: Union[str, Path], tokenizer: Optional[Callable[[str], str]] = None
+) -> int:
+    counter = 0
+    with open(file_path) as file:
+        if tokenizer:
+            for _item in tokenizer(file.read()):
+                counter += 1
+        else:
+            for _line in file:
+                counter += 1
+        return counter
```

Comment:
What is the right way? Leave this `return` inside `with` context manager, outside, or it doesn't matter in this case?

Reply:
It is not about the return.  Are you doing something almost the same?

Reply:
I came by this oneliner idea...

Reply:
Yes, it is. My idea was to write a default tokenizer.

Reply:
I got mentally fixated on that `None` from the initial template...

Reply:
And it is still should be None. Because you need to handle None by task requirements. Default tokenizer should be inside a function not as a part of its API.